### PR TITLE
feat: Nx segment

### PR DESCRIPTION
**Pull request by evanfuture**
### Prerequisites

- [x] I have read and understood the [contributing guide][CONTRIBUTING.md]
- [x] The commit message follows the [conventional commits][cc] guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added/updated (for bug fixes/features)

### Description

Add a new segment to show the Nx (monorepo tooling) version.

### CODE REVIEW

File: src/segments/nx.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+func (a *Nx) Init(props properties.Properties, env environment.Environment) {
+	a.language = language{
+		env:        env,
+		props:      props,
+		extensions: []string{"workspace.json"},
+		commands: []*cmd{
+			{
+				regex: `(?:(?P<version>((?P<major>[0-9]+).(?P<minor>[0-9]+).(?P<patch>[0-9]+))))`,
+				getVersion: func() (string, error) {
```
Comment (JanDeDobbeleer) [inline src/segments/nx.go]: Can you define this function below and reference it here?

**Review by JanDeDobbeleer — CHANGES_REQUESTED** [review]

File: src/segments/nx.go (hunk 1 above)
Comment (evanfuture) [reply src/segments/nx.go]: Done.  I'm not really proficient with Go, but it seems to work.

I basically just copy/pasted from the Angular segment. 😅

**Comment by JanDeDobbeleer** [comment]
@evanfuture there's some other changes needed but I might just do them as the logic isn't flawed, it just needs to be a little more cross platform proof (and I'm not at my laptop right now).

File: src/segments/nx.go (hunk 1 above)
Comment (JanDeDobbeleer) [reply src/segments/nx.go]: I noticed that only after the facts. And truth be told, that one was actually not OK. I must've been a bit under the weather when I merged that one in. But, fixed it.

**Pull request merged**